Source/Core/Core/PowerPC/Interpreter/Interpreter_FPUtils.h

```diff
@@ -25,53 +25,38 @@ enum class FPCC
   FU = 1,  // ?
 };
 
-inline void SetFPException(u32 mask)
+inline void SetFPException(UReg_FPSCR* fpscr, u32 mask)
 {
-  if ((FPSCR.Hex & mask) != mask)
+  if ((fpscr->Hex & mask) != mask)
   {
-    FPSCR.FX = 1;
+    fpscr->FX = 1;
   }
 
-  FPSCR.Hex |= mask;
-  FPSCR.VX = (FPSCR.Hex & FPSCR_VX_ANY) != 0;
+  fpscr->Hex |= mask;
+  fpscr->VX = (fpscr->Hex & FPSCR_VX_ANY) != 0;
 }
 
-inline void SetFI(int FI)
+inline void Helper_UpdateCR1(PowerPC::PowerPCState* ppcs)
 {
-  if (FI)
-  {
-    SetFPException(FPSCR_XX);
-  }
-  FPSCR.FI = FI;
-}
-
-inline void UpdateFPSCR()
-{
-  FPSCR.VX = (FPSCR.Hex & FPSCR_VX_ANY) != 0;
-  FPSCR.FEX = (FPSCR.VX & FPSCR.VE) | (FPSCR.OX & FPSCR.OE) | (FPSCR.UX & FPSCR.UE) |
-              (FPSCR.ZX & FPSCR.ZE) | (FPSCR.XX & FPSCR.XE);
-}
-
-inline void Helper_UpdateCR1()
-{
-  PowerPC::ppcState.cr.SetField(1, (FPSCR.FX << 3) | (FPSCR.FEX << 2) | (FPSCR.VX << 1) | FPSCR.OX);
+  ppcs->cr.SetField(1, (ppcs->fpscr.FX << 3) | (ppcs->fpscr.FEX << 2) | (ppcs->fpscr.VX << 1) |
+                           ppcs->fpscr.OX);
 }
 
-inline double ForceSingle(double value)
+inline double ForceSingle(bool flush_to_zero, double value)
```

Comment:
This should just pass the FPSCR by const reference and use it within the function. This way the caller doesn't need to constantly care about passing the correct flag.